app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -75,13 +121,40 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
                     it.setOnClickListener(this)
                 }
 
+        uiLifecycleCancelJob = Job()
+
+        initTiles()
         setupUrlInput()
         turboButton.isChecked = Settings.getInstance(context).isBlockingEnabled
         navButtonSettings.setImageResource(R.drawable.ic_settings) // Must be set in code for SVG to work correctly.
 
         val tintDrawable: (Drawable?) -> Unit = { it?.setTint(ContextCompat.getColor(context, R.color.tv_white)) }
         navCloseHint.compoundDrawablesRelative.forEach(tintDrawable)
         navUrlInput.compoundDrawablesRelative.forEach(tintDrawable)
+
+        tileContainer.addOnScrollListener(object : RecyclerView.OnScrollListener() {
```

Comment:
As mentioned earlier, move this whole recyclerview code into a different file - this is already nicely modular, no reason to glop it into a big file.